3.preprocessing_features/scripts/1.sc_quality_control.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Perform single-cell quality control
+# 
+# In this notebook, we perform single-cell quality control using coSMicQC. We filter the single cells by identifying outliers with z-scores, and use either combinations of features or one feature for each condition. We use features from the AreaShape and Intensity modules to assess the quality of the segmented single-cells:
+# 
+# ### Assessing poor nuclei segmentation
+# 
+# Due to high confluence with various seeding densities, sometimes nuclei overlap on top of each other, creating highly intense clusters within the Hoechst channel. To identify these nuclei, we use:
+# 
+# - **Nuclei Area:** This metric quantifies the number of pixels in a nucleus segmentation. 
+# We detect nuclei that are abnormally large, which likely indicates poor nucleus segmentation where overlapping nuclei are merged into one segmentation. 
+# - **Nuclei Intensity:** This metric quantifies the total intensity of all pixels in a nucleus segmentation. 
+# In combination with abnormally large nuclei, we detect nuclei that are also highly intense, likely indicating that this a group of overlapped nuclei.
+# 
+# For the preliminary dataset, we are working with cells that have not been treated so we do not expect any crazy phenotypes. Given that context, we can use a feature called Solidity. From ChatGPT, the simple explanation is that this features compares the area of the object to its convex hull, which measures compactness in relation to convexity. High solidity implies few indentations, while lower solidity indicates more irregularity.
+# 
+# - **Nuclei Solidity:** This metric quantifies the compactness of the nuclei shape.
+# When a nuclei is mis-segmented, we see more protrusions or harsh outlines around the segmentations, which we expect this is what this feature will detect.
+
+# In[1]:
+
+
+import dask.dataframe as dd
+import pathlib
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import gc
+import time
+
+from cosmicqc import find_outliers
+
+
+# # Set functions for plotting
+
+# ### Function to plot scatterplot
+
+# In[2]:
+
+
+def plot_large_nuclei_outliers(
```

Comment:
Fair point! Since I am ever only going to use these functions in that notebook, I decided to just keep the functions there, but I might decide to refactor this later. Thanks for bringing this up!